rust/rbac-registration/src/cardano/cip509/validation.rs

```diff
@@ -108,160 +99,20 @@ pub(crate) fn validate_txn_inputs_hash(
-                                                                match name.gn_value() {
-                                                                    GeneralNameValue::Text(s) => {
-                                                                        if let Ok(uri) = Cip0134Uri::parse(s) {
-                                                                            if let Address::Stake(a) = uri.address() {
-                                                                                pk_addrs.push(a.payload().as_hash().to_vec());
-                                                                            }
-                                                                        }
-                                                                    },
-                                                                    _ => {
-                                                                        validation_report.push(
-                                                                            format!("{function_name}, Failed to get the value of subject alternative name"),
-                                                                        );
-                                                                    }
-                                                                }
-                                                            }
-                                                        }
-                                                    },
-                                                    c509_certificate::extensions::alt_name::GeneralNamesOrText::Text(_) => {
-                                                        validation_report.push(
-                                                            format!("{function_name}, Failed to find C509 general names in subject alternative name"),
-                                                        );
-                                                    }
-                                                }
-                                            },
-                                            _ => {
-                                                validation_report.push(
-                                                    format!("{function_name}, Failed to get C509 subject alternative name")
-                                                );
-                                            }
-                                        }
-                            }
-                        }
-                    },
-                    _ => continue,
-                }
-            }
-        }
-    } else {
-        validation_report.push(format!("{function_name}, Unsupported transaction era"));
-        return None;
-    }
+    let addresses = match Cip0134UriList::new(cip509, txn) {
```

Comment:
I would rather put a validation about the `txn` type at this place, before building `Cip0134UriList` it will be more explicit I think

Reply:
Makes sense, fixed.